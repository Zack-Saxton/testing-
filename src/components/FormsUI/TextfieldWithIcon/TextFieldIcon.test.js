import { cleanup, render, fireEvent } from '@testing-library/react';
import React from 'react';
import TextFieldWithIcon from './index.js';

afterEach(cleanup);
const component = () =>{
    return(<TextFieldWithIcon
        name="userName1"
        label="Enter Username"
        icon="cloud"
        iconColor="#595E6E"
        iconPosition="right"
        materialProps={ { "data-testid": "test" } }
        required={ true }
    />);
}

test("Check the text filed is render", () => {
    const container = render(component());
    const input = container.getByTestId('test');
    expect(input).toBeTruthy();
});

test("Check the text filed is render with Icon", () => {
    const container = render(component());
    const input = container.getByTestId('icon');
    expect(input).toBeTruthy();
});

test("Check can able to enter value", () => {
    const container = render(component());
    const input = container.getByTestId('test');
    fireEvent.change(input, { target: { value: "Mariner" } });
    expect(input.value).toBe('Mariner');
});
  

test('should match the snapshot', () => {
    const { asFragment } = render(component());
    expect(asFragment).toMatchSnapshot();
});