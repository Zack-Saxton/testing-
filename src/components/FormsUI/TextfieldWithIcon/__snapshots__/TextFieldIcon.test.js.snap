// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      style="display: inline-flex;"
    >
      <div
        className="MuiFormControl-root MuiTextField-root makeStyles-dynamic-15 MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root makeStyles-cssLabel-14 MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink="false"
        >
          Enter Username
          <span
            aria-hidden="true"
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
            â€‰*
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            className="MuiInputBase-input MuiInput-input"
            data-test-id="test"
            name="userName1"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="MuiGrid-root"
        style="padding-top: 20px; padding-left: 10px;"
      >
        <span
          aria-hidden="true"
          className="material-icons MuiIcon-root"
          data-test-id="icon"
        >
          cloud
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
